daemon/emer-machine-id-provider.c

```diff
@@ -414,7 +433,7 @@ write_tracking_id_file (const gchar  *path,
         }
     }
 
-  if (!g_file_set_contents (path, serialized_override_machine_id, -1, error))
+  if (!g_file_set_contents (path, dehyphenated_serialized_machine_id, -1, error))
```

Comment:
Why didn't the test suite catch this bug? Can we add a test that would have failed before, and now passes?